### PR TITLE
tracker2: handle nonexistent users better

### PR DESCRIPTION
**Pull request by jakob223**
Previously, the app would show a loading screen forever at the user's profile. Now it shows the standard deeplink error modal.

I think the error modal could probably be better, so I filed a design ticket DESIGN-2944.

cc @keybase/y2ksquad @zapu

### CODE REVIEW

File: shared/actions/tracker2.tsx [hunk 1]
```diff
@@ -303,10 +305,20 @@ const loadNonUserProfile = async (action: Tracker2Gen.LoadNonUserProfilePayload)
     return false
   } catch (e) {
     logger.warn(`Error loading non user profile: ${e.message}`)
-    return false
+    return Tracker2Gen.createErrorLoadingNonUserProfile({assertion, error: e.message})
```
Comment (zapu) [inline shared/actions/tracker2.tsx]: make sure to test this... last time I checked, GetNonUserDetails also bailed with error on keybase usernames, and `loadNonUserProfile` seems to be always called, even if you are not attempting to view sbs profile.

maybe my findings were wrong though

Comment (jakob223) [reply shared/actions/tracker2.tsx]: Good catch

**Comment by jakob223** [comment]
Untagging review while I make this not trigger on literally every profile load ever

File: shared/actions/tracker2.tsx [hunk 2]
```diff
@@ -332,6 +348,7 @@ function* tracker2Saga() {
   yield* Saga.chainAction2(EngineGen.keybase1NotifyUsersUserChanged, refreshSelf)
   yield* Saga.chainAction(Tracker2Gen.loadNonUserProfile, loadNonUserProfile)
   yield* Saga.chainAction(Tracker2Gen.updatedDetails, refreshTrackerBlock)
+  // yield* Saga.chainAction(Tracker2Gen.errorLoadingNonUserProfile, showBadUsernameDeeplinkError)
```
Comment (songgao) [inline shared/actions/tracker2.tsx]: 🚿

**Pull request merged**